### DISCUSSION TITLE
Running just one unit test, not the whole suite

### DISCUSSION THREAD

**Discussion by sethpeasley** [tui-cs/Terminal.Gui#1297] [Q&A] [answered]
Hello! 

I'm trying to add some unit tests to GUI.CS. To that end, I want to focus on running just the test (or some small subset of tests) that I'm currently working on, instead of the whole test suite. I can do this from the command line with `dotnet test --filter <filter goes here>`. I've verified, I can use this technique on two other .NET project on my machine. For example, to run just the method "JustForTesting()": `dotnet test -- filter JustForTesting`. This test will execute, and be the only test case to run. 

Unfortunately, when I try it with GUI.CS Unit Test suite, all the tests run, every time. I've compared the .csproj files and they seem all pretty similar. I'm not sure what's causing this or how to just run one test. Can you folks advise on how to achieve this?

I'm running Kubuntu 18.04, with .NET SDK 5.0.202. I'm not sure what other system info might be helpful.

Thanks!

**Comment by tig** [comment]
Hi Seth,

I've never used `dotnet test --filter` before. I just tried it and it seems to work:

![](https://i.imgur.com/e9UxwBY.png)

That was on Windows. 

This is Debian (WSL):
![](https://i.imgur.com/uydnqKO.png)

Odd.

**Comment by tznind** [comment, marked as answer]
> [...] For example, to run just the method "JustForTesting()": `dotnet test -- filter JustForTesting`. [...]

In your comment you have a space between `--` and `filter` (I assume that's a typo? - there shouldn't be one).

**Reply by sethpeasley** [reply]
I am both relieved this solves the issue but slightly embarrassed I didn't catch that. The command is executing as I expect now. I suspect I got the `--flag` and `-- [args]` operators confused in my brain. Thanks for the help!
